### PR TITLE
Support for Pull requests

### PR DESCRIPTION
**Pull request by thomaspaulmann**
As promised, here is the second PR for the support of Pull Requests. PRs and Issues are mostly the same in GitHub, so it's a little bit ugly to separate them. Anyways, I've added some methods to separate the access to PRs or Issues. It's more or less only a filter for the optional PR-JSON-Node. Please have a look and let me know what do you think about it.
An other solution is only to attach the optional Pull Request member to the Issue-Model and the programmer could decide how and what he want.

### CODE REVIEW

**Comment by thomaspaulmann** [comment]
Furthermore I improved the previous PR #34  a little bit and make the author of the issue not optional, the assignee really optional and added the HTML url to the issue itself. Have fun :smile_cat:

**Comment by nwest** [comment]
@thomaspaulmann could you rebase this and add the files to the project for all of the targets?

File: OctoKit/Issue.swift [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    public let id: Int
+    public var url: String?
+    public var htmlUrl: String?
+    public var number: Int?
+    public var title: String?
+    public var body: String?
+    public var state: String?
+    public var user: User
+    public var assignee: User?
+    public var pullRequest: PullRequest?
```
Comment (pietbrauer) [inline OctoKit/Issue.swift]: I am not sold on this tbh. I would treat the fact that PullRequests and issues are nearly identical as a coincidence and treat them as separate things and therefore separate models.

**Comment by pietbrauer** [comment]
This also conflicts with master now due to #36

**Pull request closed without merging**